### PR TITLE
Converts to using npm over yarn 1

### PR DESCRIPTION
**Pull request by scalvert**
This is something @rwjblue and I have been discussing lately, and in fact we're moving as many of our OSS repos to using npm now over yarn 1. We're doing this for the following reasons:

- yarn 1 is no longer maintained, and bugs that exist in yarn 1 aren't prioritized to fix
- yarn used to be faster, but npm (as of v7) is blazing fast
- yarn was the only one to support workspaces, but now npm supports them to (as of v7). This is unrelated to this repo, but is a _thing_.
- yarn's registry will often timeout, causing frustrating re-runs of CI jobs. Example: https://github.com/ember-template-lint/ember-template-lint/runs/6412481173?check_suite_focus=true 

Once this change lands, contributors should follow these steps:

1. `rm -rf node_modules`
2. `npm i`
3. `npm test` to ensure things are working.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -100,6 +100,6 @@ jobs:
       - uses: volta-cli/action@v1
 
       - name: install dependencies
-        run: yarn install --no-lockfile
+        run: npm install --no-package-lock
```
Comment (bmish) [inline .github/workflows/ci.yml:103]: `npm ci`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,13 +22,13 @@ jobs:
       - uses: volta-cli/action@v1
 
       - name: install dependencies
-        run: yarn install --frozen-lockfile
+        run: npm install --frozen-lockfile
```
Comment (bmish) [inline .github/workflows/ci.yml:25]:
```suggestion
        run: npm ci
```

**Review by bmish — COMMENTED** [review]
We need to add `engine-strict=true` to `.npmrc` to ensure that installing dependencies still enforces the Node engine check since NPM doesn't do this by default (to ensure we only use dependencies that are compatible with our supported Node versions, more info in https://github.com/release-it/release-it/pull/890).

https://docs.npmjs.com/cli/v8/using-npm/config#engine-strict

File: .github/workflows/ci.yml (hunk 1 above)
Comment (scalvert) [reply .github/workflows/ci.yml:103]: Oops, missed that one!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (scalvert) [reply .github/workflows/ci.yml:25]: Missed this one too!

Comment (scalvert) [reply .github/workflows/ci.yml]: Fixed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (scalvert) [reply .github/workflows/ci.yml]: Fixed.

**Comment by bmish** [comment]
@scalvert we should update that job to use `npm ci --force` to ignore the engine check only in Node 12. That way we can have `engine-strict=true` to `.npmrc`.

**Comment by scalvert** [comment]
> @scalvert we should update that job to use `npm ci --force` to ignore the engine check only in Node 12. That way we can add `engine-strict=true` to `.npmrc` applying everywhere else.

Nice, was just looking for this. Thanks!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -100,6 +100,6 @@ jobs:
       - uses: volta-cli/action@v1
 
       - name: install dependencies
-        run: yarn install --no-lockfile
+        run: npm ci
```
Comment (bmish) [inline .github/workflows/ci.yml:103]: For the floating dependencies job, we specifically need to make sure it runs without a lockfile, try `npm ci --no-package-lock`.

Comment (scalvert) [reply .github/workflows/ci.yml:103]: I thought I'd added that. WTF is going on...

File: package.json [hunk 4]
```diff
@@ -26,9 +26,9 @@
   "scripts": {
     "lint": "npm-run-all --continue-on-error --parallel lint:*",
     "lint:docs": "markdownlint \"**/*.md\"",
-    "lint:docs:fix": "yarn lint:docs --fix",
+    "lint:docs:fix": "npm run lint:docs --fix",
```
Comment (bmish) [inline package.json:29]: To pass options to these NPM commands, I believe we need `-- --fix` for this to work...

Comment (scalvert) [reply package.json:29]: Another good catch.

Comment (scalvert) [reply package.json]: Fixed.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (scalvert) [reply .github/workflows/ci.yml]: Fixed.

**Review by bmish — APPROVED** [review]

**Pull request merged**